src/tcache.c

```diff
@@ -963,6 +985,55 @@ thread_tcache_max_set(tsd_t *tsd, size_t tcache_max) {
 	assert(tcache_nbins_get(tcache_slow) == sz_size2index(tcache_max) + 1);
 }
 
+bool
+tcache_bins_ncached_max_write(tsd_t *tsd, char *settings, size_t len) {
+	tcache_t *tcache = tsd_tcachep_get(tsd);
+	bool enabled = tcache_available(tsd);
+	assert(tcache != NULL);
+	cache_bin_info_t tcache_bin_info[TCACHE_NBINS_MAX];
+	tcache_bin_settings_backup(tcache, tcache_bin_info);
+	const char *bin_settings_segment_cur = settings;
+	size_t len_left = len;
+	assert(len_left != 0);
+
+	do {
+		size_t size_start, size_end;
+		size_t ncached_max;
+		bool err = multi_setting_parse_next(&bin_settings_segment_cur,
+		    &len_left, &size_start, &size_end, &ncached_max);
+		if (err) {
+			return true;
+		}
+		if (size_end > TCACHE_MAXCLASS_LIMIT) {
+			size_end = TCACHE_MAXCLASS_LIMIT;
+		}
+		szind_t bin_start = sz_size2index_compute(size_start);
+		szind_t bin_end = sz_size2index_compute(size_end);
+		for (szind_t i = bin_start; i <= bin_end; i++) {
+			/* No caps are set for manually set ncached_max. */
```

Comment:
Well, there is still a width limit:
https://github.com/jemalloc/jemalloc/blob/005f20aa7fdef1be6f9fe46e4f2e7b88177a9f21/include/jemalloc/internal/cache_bin.h#L24

So we need to check and limit the ncached_max (instead of force cast).

Reply:
By limit, I think you mean we should return `EINVAL` to let the users know this value is not valid, right?

Reply:
Yeah cache_bin_sz_t is uint16_t (the references in the link should be changed to cache_bin_sz_t). Making it larger would cause the cache_bin struct to use more bytes in the hot cachelines, so we probably will not consider that move, as you mentioned it covers up to 8K already.